plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -680,4 +772,76 @@ public void setColumnComment(ConnectorSession session, JdbcTableHandle handle, J
                 comment.orElse(""));
         execute(session, sql);
     }
+
+    private static class StatisticsDao
+    {
+        private final Handle handle;
+
+        public StatisticsDao(Handle handle)
+        {
+            this.handle = requireNonNull(handle, "handle is null");
+        }
+
+        Long getRowCount(String schema, String tableName)
+        {
+            return handle.createQuery("SELECT NUM_ROWS FROM ALL_TAB_STATISTICS WHERE OWNER = :schema AND TABLE_NAME = :table_name and PARTITION_NAME IS NULL")
+                    .bind("schema", schema)
+                    .bind("table_name", tableName)
+                    .mapTo(Long.class)
+                    .findFirst()
+                    .orElse(null);
+        }
+
+        List<ColumnStatisticsResult> getColumnStatistics(String schema, String tableName)
+        {
+            // [SEP-3425]    we are not using ALL_TAB_COL_STATISTICS, here because we observed queries which took multiple minutes when obtaining statistics for partitioned tables.
```

Comment:
The linked issue seems to be resolved and all necessary context exists in this code comment - so I'd drop that external reference.